x/nft/codec.proto

```diff
@@ -12,7 +12,7 @@ message NonFungibleToken {
 }
 
 message ActionApprovals {
-    string action = 1;
+    string action = 1 [(gogoproto.customtype) = "Action", (gogoproto.nullable) = false];
```

Comment:
Did you check this works in other compilers?
Eg. protoc to generate js (I assume all gogoproto options would be ignored there)

Reply:
I am expecting that this works **only** for Go compiler. Other compiles will create this as a `string` type. This is ok, because `Action` is an alias to the string type (otherwise it would not work anyway).

It is possible to define custom marshal/unmarshal methods on custom type and have for example a JSON serialized field. This is an awesome extension!